src/memote/metabolites.jl

```diff
@@ -0,0 +1,120 @@
+#=
+This file contains a collection of tests based on Memote. See Lieven, C., Beber,
+M.E., Olivier, B.G. et al. MEMOTE for standardized genome-scale metabolic model
+testing. Nat Biotechnol 38, 272–276 (2020).
+https://doi.org/10.1038/s41587-020-0446-y for details.
+=#
+
+"""
+$(TYPEDSIGNATURES)
+
+Return a list of all boundary reactions that allow flux into the model and
+create a metabolite. Assume that boundary reactions only create a single
+metabolite. Set the kwarg `only_into=false` to also return metabolites that can
+be produced by the model under default conditions.
+"""
+function metabolites_medium_components(model; only_into=true)
+    mets = String[]
+    for (rid, lb, ub) in zip(reactions(model), bounds(model)...)
+        if is_boundary(model, rid)
+            mid = first(keys(reaction_stoichiometry(model, rid))) 
+            only_into && lb < 0 && ub <= 0 && push!(mets, mid)
+            !only_into && lb < 0 && push!(mets, mid)
+        end
+    end
+    mets
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+List all metabolites without a formula. Use `corner_cases` to specify an extra
+case to check for formula's that are not properly assigned.
+"""
+metabolites_no_formula(model; corner_cases = ["X", "x", ""]) = [mid for mid in metabolites(model) if isnothing(metabolite_formula(model, mid)) || metabolite_formula(model, mid) in corner_cases]
+
+"""
+$(TYPEDSIGNATURES)
+
+List all metabolites without a charge. Use `corner_cases` to specify an extra
+case to check for charge's that are not properly assigned.
+"""
+metabolites_no_charge(model; corner_cases = []) = [mid for mid in metabolites(model) if isnothing(metabolite_charge(model, mid)) || metabolite_charge(model, mid) in corner_cases]
+
+"""
+$(TYPEDSIGNATURES)
+
+Test if metabolites `m1` and `m2` are different by comparing their
+`test_annotation` fields in the annotations of each metabolite. Note, if no
+annotations are present for one or both of the metabolites, then return `true`. 
+"""
+function metabolites_are_duplicated(model, m1, m2; test_annotation = "inchi_key")
```

Comment:
this should check compartments too (you can have internal and external water without much of a big problem).

Again, do not default the `inchi_key` here. It also makes sense to check CHEBI IDs etc.

Reply:
it already does the compartment specific one, and the inchi key is the memote.py default but change be changed by setting the master parameter struct